src/lib/qnio/nio_server.c

```diff
@@ -470,6 +497,64 @@ qns_server_start(char *node, char *port)
                 }
                 continue;
             }
+            else if (qns_ctx->activefds[i].events & EPOLLIN)
+            {
+                int             infd;
+                int             sslerr;
+                if(!cmn_ctx->ssl_ctx)
+                {
+                    /* why did we get here? */
+                    nioDbg("Unknown error");
+                    return (-1);
+                }
+                /* complete ssl accept */
+                nioDbg("Completing ssl accept");
+                ep = (struct endpoint *)qns_ctx->activefds[i].data.ptr;
+                infd = ep->sock;
+                ssl = ep->ssl;
+                epoll_ctl(qns_ctx->epoll_fd, EPOLL_CTL_DEL,
+                          infd, &event);
+                ret = SSL_accept(ssl);
+                if (ret <= 0) {
+                    nioDbg("ssl accept failed %d", ret);
+                    /* Maybe try SSL_accept again */
+                    sslerr = SSL_get_error(ssl, ret);
+                    nioDbg("ssl error is %d", sslerr);
+                    if (sslerr == SSL_ERROR_WANT_READ || sslerr == SSL_ERROR_WANT_WRITE) 
+                    {
+                        nioDbg("ssl accept still not complete");
+                        continue;
```

Comment:
You are right. EPOLL_CTL_DEL needs to happen after successful SSL_accept. Will fix this.